packages/account/src/account.ts

```diff
@@ -542,13 +543,15 @@ export class Account extends AbstractAccount {
 
     const findAssetInput = (assetId: string) =>
       txRequestClone.inputs.find((input) => {
-        if ('assetId' in input) {
+        if (input.type === InputType.Coin) {
           return input.assetId === assetId;
         }
-        if ('recipient' in input) {
+
+        // We only consider the message input if it has no data.
+        // Messages with `data` cannot fund the gas of a transaction.
+        if (input.type === InputType.Message && bn(input.data).isZero()) {
```

Comment:
Should we use the helper `isRequestInputMessageWithoutData` here?

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3078/commits/06a2fbc36d40324be0689f2af01fa0a1263ffdbe